src/lab12.py

```diff
@@ -283,7 +283,8 @@ def click(self, x, y):
                     if elt.tag == "form" and "action" in elt.attributes:
```

Comment:
Ok, now that I understand the issue, what if we instead change the `while elt` above this to be `while elt.parent`. My point here is that the `if` statement in the outer `while` loop is kind of weird given the condition of that loop. It's really a problem caused by this inner `while` loop so I'd rather fix it there.

Reply:
Good idea.